app/helpers/argo_helper.rb

```diff
@@ -102,11 +102,11 @@ def render_buttons(doc, object = nil)
     end
 
     # if this is an apo and the user has permission for the apo, let them edit it.
-    if (object.datastreams.include? 'roleMetadata') && (current_user.is_admin || current_user.is_manager || object.can_manage_item?(current_user.roles(apo_pid)))
+    if object.datastreams.include?('roleMetadata') && can?(:manage_item, object)
       buttons << {:url => url_for(:controller => :apo, :action => :register, :id => pid), :label => 'Edit APO', :new_page => true}
       buttons << {:url => url_for(:controller => :apo, :action => :register_collection, :id => pid), :label => 'Create Collection'}
     end
-    if object.can_manage_item?(current_user.roles(apo_pid)) || current_user.is_admin || current_user.is_manager
+    if can?(:manage_item, object)
       buttons << {
```

Comment:
Indeed, and cancan makes this distinction in the `Ability` class, instead of forcing us to replicate it in all other contexts.